Wordle.Core/SquareWord.cs

```diff
@@ -0,0 +1,45 @@
+namespace Wordle.Core;
+
+public class SquareWord : BaseGame
+{
+
+    public char[,] Grid { get; }
+    public IReadOnlyList<string> SelectedWords { get; }
+    public SquareWord(int wordLength, int possibleTries, WordList list)
+        : base(wordLength, possibleTries, false, list)
+    {
+        Grid = new char[WordLength, WordLength];
+        do
+        {
+            var selectedWords = new List<string>(WordLength);
+            var firstVerticalWord = SelectRandomWord();
+            for (var y = 0; y < WordLength; y++)
+            {
+                var word = SelectWordStartingWith(firstVerticalWord[y]);
+                selectedWords.Add(word);
+                for (var x = 0; x < wordLength; x++)
+                    Grid[x, y] = word[x];
+            }
```

Comment:
Way too slow
Try to:
1. Add vertical word
2. Add corresponding horizontal word
3. Add corresponding vertical word
4. Repeat step 2